### ISSUE TITLE
Object Allocation docs out of date?

### ISSUE THREAD

**Issue by Tokazama** [JuliaLang/julia#48784]
I think that the docs for object allocation are out of date. For example, the docs say mention most objects going through `jl_gc_allocobj`...

https://github.com/JuliaLang/julia/blob/f0eadd076f2728797cf5be88b61cda1d9f1dd515/doc/src/devdocs/object.md?plain=1#L189

but the only place it's used outside of docs and the function definition is in a test.

**Comment by inkydragon** [comment]
> I think that the docs for object allocation are out of date.

Yes, it looks like devdoc hasn't been updated in a long time (> 5 yr).

The function `jl_gc_alloc` is more commonly used:
https://github.com/JuliaLang/julia/blob/f0eadd076f2728797cf5be88b61cda1d9f1dd515/src/julia_internal.h#L477

**Comment by TakshDhabalia** [comment]
made a small commit for the same , open to suggestions.
